proposals/js-promise-integration/Overview.md

```diff
@@ -72,16 +83,34 @@ fetch('demo.wasm').then(response =>
 });
 ```
 
-In this example, we have a WebAssembly module that is a very simplistic state machine&mdash;every time you update the state, it simply calls an import to compute a delta to add to the state.
-On the JavaScript side, though, the function we want to use for computing the delta turns out to need to be run asynchronously; that is, it returns a Promise of a Number rather than a Number itself.
+In this example, we have a WebAssembly module that is a very simple state machine&mdash; driven from JavaScript. Whenever the JavaScript client code wishes to update the state, it invokes the exported `update_state` function. In turn, the WebAssembly `update_state` function calls an import – `compute_delta` – to compute a delta to add to the state.
+
+On the JavaScript side, though, the function we want to use for computing the delta turns out to need to be run asynchronously; that is, it returns a `Promise` of a `Number` rather than a `Number` itself.
+
+We can bridge this synchrony gap by bracketing the exported `update_state` function and the imported `compute_delta` function using a common `suspender` and wrapping the functions.
+
+The `suspender.returnPromiseOnSuspend` function takes a function as argument – in this case `update_state` – and wraps it into a new function; which will be the function actually used by JavaScript code. The new function invokes the wrapped function: calls it with the same arguments and returns the same results. The difference shows up if the inner function ever suspends.
+
+The `suspender.suspendOnReturnedPromise` function also takes a function as argument: – `compute_delta` which was the original imported function. When it is called, the wrapped function calls `compute_delta` and inspects the returned result. If that result is a `Promise` then, instead of returning that `Promise` to the WebAssembly module, suspends the computation instead and `update_state` will return with an updated `Promise`.
```

Comment:
```suggestion
The `suspender.suspendOnReturnedPromise` function also takes a function as argument: `compute_delta`—the original imported function. When it is called, the wrapper calls `compute_delta` and inspects the returned result. If that result is a `Promise` then, instead of returning that `Promise` to the WebAssembly module, the wrapper suspends `suspender`'s WebAssembly computation instead.
```